packages/yoroi-extension/chrome/extension/background.js

```diff
@@ -272,10 +297,16 @@ chrome.runtime.onMessage.addListener(async (request, sender, sendResponse) => {
   }
   // alert(`received event: ${JSON.stringify(request)}`);
   if (request.type === 'connect_response') {
+    if (request.tabId == null) return;
     const connection = connectedSites.get(request.tabId);
     if (connection && connection.status != null && typeof connection.status === 'object') {
-      connection.status.resolve(request.accepted ? request.account : null);
-      connection.status = request.account;
+      if (request.accepted === true) {
+        connection.status.resolve(request.publicDeriverId);
+        connection.status = request.publicDeriverId;
+      } else {
+        connection.status.resolve(null);
+        // TODO: what to do here? Delete entry from connectedSites? Do nothing?
```

Comment:
@rooooooooob do you know how this is supposed to be handled?

Reply:
Yes, remove the entry from `connectedSites`. it will get re-added the next time the dApp tries to connect and there's no point keeping it around. It might be a good idea to also update how `'get_connected_sites'` is handled to check if the site is connected yet since it could be in `connectedSites` but have the status be pending.